### DISCUSSION TITLE
Using the declarative base, is there a way to access th mapping from outside a class instance?

### DISCUSSION THREAD

**Discussion by cnicodeme** [sqlalchemy/sqlalchemy#9571] [Usage Questions]
Hi!

I have created many Models using the declarative base, like:

```python
BaseModel = declarative_base()

class Account(BaseModel):
    id = ...

class Message(BaseModel):
     id = ...

# etc
```

Knowing that, is there a way to access a class without having to import it, but by calling it via its name, something like

```python
from sqlalchemy.something import registry

Message = registry.get('Message')
Message.query.filter(...)
```

I know I could simply import the class in the other file. The reason I'm not doing this is simply because I'm having circular imports I'd like to avoid.

Thank you :)

**Comment by cnicodeme** [comment]
So far, here's what I've done, curious to know if you have a better and more efficient recommendation:

```python
BaseModel = declarative_base()

def get_model(name: str):
    for mapper in BaseModel.registry.mappers:
        if mapper.class_.__name__ == name:
            return mapper.class_

    raise LookupError(f'Class {name} was not found.')
```

That way I can do :

```python
from db import get_model

Account = get_model('Account')
Account.query.filter(...)
```

**Reply by zzzeek** [reply]
we don't yet have a public API for this feature, it's mentioned at #6080 but is low priority.

There's Python -agnostic methods of dealing with import cycles as well as this is a problem that is bigger in scope than just SQLAlchemy.
